py-rse/style.Rmd

```diff
@@ -4,52 +4,41 @@
 source(here::here("_common.R"))
 ```
 
-To paraphrase @Dobz1973,
+Nothing in biology makes sense except in light of evolution @Dobz1973.
+Similarly,
 nothing in software development makes sense except in light of human psychology.
 This is particularly true when we look at programming style.
 Computers don't need to understand programs in order to execute them,
-but people do in order to create them, maintain them, and fix them.
-The more clearly those programs are laid out,
-the easier it is to find things and make sense of them.
-This lesson therefore describes some widely-used rules of Python programming style,
-and some features of the language that you can use to make your programs more flexible and more readable.
-
-The biggest benefit of having a second person work on a programming project
-is therefore not getting twice as much code written,
-but having code reviewed.
-Study after study over more than 40 years has shown that code review is the most effective way to find bugs in software
-@Faga1976,@Faga1986,@Cohe2010@Bacc2013.
-Despite this,
-it still isn't common in research software development,
-in part because it isn't part of the culture @Sega2005,
-but also because code review is mostly useful
-when the reviewers understand the problem domain well enough to comment on algorithms and design choices
-rather than indentation and variable naming,
-and the number of people who can do that for a research project is typically very small—sometimes
-as small as one @Petr2014.
+but people do if they are to create, debug, and extend them.
+This lesson therefore presents guidelines for good Python programming style
+and introduces some language features that can make programs more flexible and more readable.
```

Comment:
Perhaps this introductory paragraph should set the scene by saying something like the following? 

"In the preceding chapters we've written a bunch of code to analyze the word counts of classic english novels. We're going to take a step back in this chapter and discuss some of the style choices that were made in developing that code and why."

Reply:
added